numalogic/models/autoencoder/pipeline.py

```diff
@@ -319,30 +319,127 @@ def with_model(
+
+    Note:
+         This class only supports Pytorch models.
+    Args:
+        beta: regularization parameter
+        rho: sparsity parameter value
+        method: regularization method
+                        supported values include {"kl_div", "L1", "L2"}
+        model: model instance
+        seq_len: sequence length
+        loss_fn: loss function used for training
+                        supported values include {"huber", "l1", "mse"}
+        optimizer: optimizer to be used for training.
+                           supported values include {"adam", "adagrad", "rmsprop"}
+        lr: learning rate
+        batch_size: batch size for training
+        num_epochs: number of epochs for training
+        std_tolerance: determines how many times the standard deviation to be used for threshold
+        reconerr_method: method used to calculate the distance
+                                between the original and the reconstucted data
+                                supported values include {"absolute", "squared"}
+        threshold_min: the minimum threshold to use;
+                              can be used when the threshold calculated is too low
+        resume_train: parameter to decide if resume training is needed. Also,
+                              based on this parameter the optimizer state dict
+                              is stored in registry.
+
+    >>> # Example usage
+    >>> from numalogic.models.autoencoder.variants import VanillaAE
+    >>> from numalogic.models.autoencoder import SparseAEPipeline
+    >>> x_train = np.random.randn(100, 3)
+    >>> model = VanillaAE(signal_len=12, n_features=3)
+    >>> sparse_ae_trainer = SparseAEPipeline(model=model, seq_len=36, num_epochs=30)
+    >>> sparse_ae_trainer.fit(x_train)
+    """
+
     def __init__(self, beta=1e-3, rho=0.05, method="kl_div", *args, **kwargs):
+        r"""
+        Function to initialize the data member values (including the inherited parent class members) when an object of this class is created. 
+
+        Args:
+            beta: regularization parameter (Defaults to 1e-3)
+            rho: sparsity parameter value (Defaults to 0.05)
+            method: regularization method (Defaults to "kl_div")
+            model: model instance
+            seq_len: sequence length
+            loss_fn: loss function used for training
+            optimizer: optimizer to be used for training
+            lr: learning rate
+            batch_size: batch size for training
+            num_epochs: number of epochs for training
+            std_tolerance: determines how many times the standard deviation to be used for threshold
+            reconerr_method: method used to calculate the distance between the original and the reconstucted data
+            threshold_min: the minimum threshold to use
+            resume_train: parameter to decide if resume training is needed
+        """
```

Comment:
You can remove the docstring in __init__, since you already added that in the class docstring.

Reply:
Sure, done.